scripts/cart.js

```diff
@@ -0,0 +1,64 @@
+let purchaseDetail = document.querySelector('.js-purchase-detail');
+
+/* Funtion Sumar input*/
+function sumar(target) {
+  let inputElement = target.parentElement.querySelector('.js-product-add-card-number');
+  let valorActual = parseInt(inputElement.value);
+  inputElement.value = valorActual + 1;
+}
+
+/* Funtion Restar input */
+function restar(target) {
+  let inputElement = target.parentElement.querySelector('.js-product-add-card-number');
+  let valorActual = parseInt(inputElement.value);
+  if (valorActual > 1) {
+    inputElement.value = valorActual - 1;
+  };
+};
+
+purchaseDetail.addEventListener('click', (event) => {
+  console.log(event)
+  let target = event.target
+  if (target.tagName === 'BUTTON') {
+    if (target.dataset.action === 'add-to-cart') {
+      sumar(target);
+    }
+
+    if (target.dataset.action === 'substract-to-cart') {
+      restar(target);
+    }
+  };
```

Comment:
- Hay un `;` que sobra
- Usando las clases creo que saldria mas compacto esto
```suggestion
  if (target.classList.contains('js-add-unit-to-product')) {
    sumar(target);
  } else if (target.classList.contains('js-substract-unit-to-product')) {
    restar(target);
  }
```